wallet-new/test/unit/UTxO/Generator.hs

```diff
@@ -259,16 +259,17 @@ genTransaction GenTrParams{..} removeUsedInputs makeOutputsAvailable notThese =
         nextHash   <- gtsNextHash <<+= 1
         numOutputs <- lift $ choose (1, gtpMaxNumOutputs)
 
-        let fee     = gtpEstimateFee numInputs numOutputs
-            inValue = sum (map (outVal . snd) inputs)
+        let inValue = sum (map (outVal . snd) inputs)
+            gos = initOutState inValue
 
-        if inValue <= fee
+        (outputs, gos') <- lift $ (`runStateT` gos) $
```

Comment:
I don't mind necessarily, but if it would make the code significantly simpler, can use the same upper bound suggestion as above? But if there was no fixpoint calculation necessary here, it doesn't really matter so much.